provision/acc_provision/apic_provision.py

```diff
@@ -5070,9 +5070,14 @@ def kube_tn(self, flavor):
                 openshift_flavor_specific_handling(data, items, system_id, old_naming, self.ACI_PREFIX, default_provide_api,
                                                    kube_api_entries, api_filter_prefix, dns_entries, filter_prefix)
             elif flavor == "docker-ucp-3.0":
-                dockerucp_flavor_specific_handling(data, items)
+                dockerucp_flavor_specific_handling(data, items, api_filter_prefix)
             elif flavor == "RKE-1.2.3":
-                rke_flavor_specific_handling(aci_prefix, data, items, self.config["rke_config"])
+                rke_flavor_specific_handling(aci_prefix, data, items, api_filter_prefix, self.config["rke_config"])
+
+        # Not adding prometheus opflex-agent contract for managed k8s environments
+        if vmm_type != "OpenShift" and flavor != "docker-ucp-3.0" and flavor != "RKE-1.2.3":
```

Comment:
Instead of checking for exact flavor names like docker-ucp-3.0 and RKE-1.2.3, we could have a flag in these flavors which indicates that there's no need for prometheus. This way it will be easier to disable prometheus for future RKE/dockerUCP flavors.

Reply:
Will do, thanks Apoorva for checking the PR.

Reply:
Discussed with Sumit and Kiran. Since this is a static port exposed from opflex-agent, we agreed to add contracts between node<-->system and node<-->default for port 9612 for all flavors. I will re-upload PR.